### DISCUSSION TITLE
7.0.0-preview4

### DISCUSSION THREAD

**Discussion by jennyf19** [AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet#2279] [IdentityModel 7x and .NET 8]
7.0.0-preview4
=======
### Bug fixes:
- Add nullables to the properties in `WsFederationMessage`. See issue [#2240](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2240) for details.
- Fix regression where `JsonWebToken.TryGetPayloadValue()` was not compatible with dictionary types. See issue [#2246](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2246) for details.
- Fix regression where dictionary claims added to `SecurityTokenDescriptor.Claims` are no longer correctly serialized. See issue [#2245](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2245) for details.
- Fix regression with a Y2038 bug. See issue [#2261](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2261) for details.

## Performance improvements:
- Remove sync-over-async pattern with `JsonWebTokens.ValidateToken`, which when in the hot path can lead to threadpool starvation. See issue [#2253](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2253) for details.
- Perf testing using brenchmark dotnet and crank, similar to aspnetcore, to better gauge requests per second perf impacts. See issue [#2232](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2232) for details.
- Use optimistic synchronization in `JsonWebToken.Audiences`. See [PR](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2243) for details.
- Reduce allocations when enumerating over collections. See [PR](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2242) for details.

## Documentation:
- Fix description for [JWT X5tS256 field](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2252).

## Fundamentals:
- Improvements to the build script to accommodate .NET's source-build requirements. See [PR](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2211) for details.

<hr /><em>This discussion was created from the release <a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/releases/tag/7.0.0-preview4'>7.0.0-preview4</a>.</em>

**Comment by jennyf19** [comment]
@kevinchalet A lot of the issues you opened were addressed in preview4. We are thinking the next release could be the GA version `7.0.0` by end of next week (week of Sept 4). Any feedback or testing you want to do before then would be much appreciated.

Thank you.

cc: @leastprivilege @brockallen @cakescience

**Reply by kevinchalet** [reply]
I gave it a try and I can confirm most of the issues are indeed fixed.

I opened https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2281 to track a potential last issue.

**Comment by kevinchalet** [comment]
> @kevinchalet A lot of the issues you opened were addressed in preview4. We are thinking the next release could be the GA version 7.0.0 by end of next week (week of Sept 4). Any feedback or testing you want to do before then would be much appreciated.

I don't see it mentioned, but I guess https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2244 is fixed, right?

I'll give it a try and let you know how it goes.

**Reply by jennyf19** [reply]
Yes, that's fixed. Missed that one.  I'll update the release notes.
